llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -952,6 +952,19 @@ InstructionCost VPInstruction::computeCost(ElementCount VF,
   }
 
   switch (getOpcode()) {
+  case Instruction::Select: {
+    // TODO: It may be possible to improve this by analyzing where the
```

Comment:
FWIW if there's still crashes, would it help to only add a cost for instructions with an underlying value? I did this in #154468 and was able to avoid any assertions on AArch64/RISC-V on llvm-test-suite

Reply:
One place on RISC-V where the underlying instruction makes a difference is when we use it to create the EVL tail folding AVL in https://github.com/llvm/llvm-project/pull/152707#discussion_r2269261910

It's not costed for in the legacy cost model so that might cause a mismatch. And from what I remember looking at this yesterday, out-of-loop reductions have a VPInstruction select inserted in the latch that isn't costed in the legacy cost model:

https://github.com/llvm/llvm-project/blob/0fa6fdfbb8652c0311d27333934cc4e102df1df2/llvm/lib/Transforms/Vectorize/LoopVectorize.cpp#L9164-L9173

I think that might now be costed with this patch? 

Partial reductions also create a select, but I think any plan with a VPPartialReduction is added to the assertion-skip-list thing.